notebook/notebookapp.py

```diff
@@ -213,6 +213,8 @@ def init_settings(self, jupyter_app, kernel_manager, contents_manager,
         
         root_dir = contents_manager.root_dir
         home = os.path.expanduser('~')
+        if sys.version_info.major == 2:
+            home = home.decode(sys.getfilesystemencoding())
```

Comment:
I believe we have a `py3compat.str_to_unicode(s, encoding=sys.getfilesystemencoding())`

Related question, though: does `os.path.expanduser(u'~')` do the right thing here, deferring to Python's on path-encoding logic?